File: src/main/java/com/govtech/assignment/service/SessionService.java
package com.govtech.assignment.service;

import java.util.Set;

import com.govtech.assignment.entity.Restaurant;
import com.govtech.assignment.entity.Session;
import com.govtech.assignment.request.CreateSessionInviteRequest;
import com.govtech.assignment.request.CreateSessionRequest;
import com.govtech.assignment.request.UpdateSessionRequest;
import com.govtech.assignment.response.CreateSessionInvitesResponse;
import com.govtech.assignment.response.CreateSessionResponse;
import com.govtech.assignment.response.SessionResponse;
import com.govtech.assignment.response.SessionsResponse;
import com.govtech.assignment.response.UpdateSessionResponse;
import com.govtech.assignment.response.UserResponse;

public interface SessionService {

	CreateSessionResponse create(CreateSessionRequest request);

	SessionsResponse find(Integer pageOffset, Integer pageLimit, String searchText, String sortBy, String sortOrder);

	SessionResponse getById(String sessionId);

	CreateSessionInvitesResponse invite(String sessionId, CreateSessionInviteRequest request);

	UpdateSessionResponse update(String sessionId, UpdateSessionRequest request);

	Boolean delete(String id);

	Set<UserResponse> getInvitees(String id);
	
	Session findById(String id);

	void suggestRestaurantToSession(Restaurant restaurant, Session session);

}

File: src/main/frontend/govtech-assignment-ui/src/app/providers/session/session.service.ts
import { Injectable } from '@angular/core';
import { RestClientService } from '../rest-client/rest-client.service';

@Injectable({
  providedIn: 'root'
})
export class SessionService {

  private baseUrl: string;

  constructor(private restClientService: RestClientService) { 
    this.baseUrl = 'api/v1/session';
  }

  create(createSessionRequest:any){
    return this.restClientService.post(createSessionRequest,this.baseUrl);
  }

  find(pageOffset:number,pageLimit:number,searchText:string,sortBy:string,sortOrder:string) {
    searchText=searchText=="" || !searchText?"ALL":searchText;
    return this.restClientService.get(this.baseUrl+"/"+pageOffset+"/"+pageLimit+"/"+searchText+"/"+sortBy+"/"+sortOrder);
  }
  
  getById(id:string) {
    return this.restClientService.get(this.baseUrl+"/"+id);
  }

  delete(id:string) {
    return this.restClientService.delete(this.baseUrl+"/"+id);
  }

  update(id:string,updateSessionRequest:any){
    return this.restClientService.put(updateSessionRequest,this.baseUrl+"/"+id);
  }

  inviteUsers(id:string,inviteUsersRequest:any){
    return this.restClientService.post(inviteUsersRequest,this.baseUrl+"/invite-users/"+id);
  }

}
